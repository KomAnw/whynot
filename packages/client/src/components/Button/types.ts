import { ButtonHTMLAttributes, ReactNode } from 'react';

export type ButtonProps = {
  variant: 'primary' | 'secondary';
  children: ReactNode;
  type?: ButtonHTMLAttributes<HTMLButtonElement>['type'];
  onSubmit?: () => void;
  onClick?: () => void;
};

export type ButtonComponent = ButtonProps & ButtonHTMLAttributes<HTMLButtonElement>;
