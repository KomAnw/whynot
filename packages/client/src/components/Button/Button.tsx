import styled from 'styled-components';
import type { ButtonComponent, ButtonProps } from './types';

export const Button = styled(
  ({ type, variant = 'primary', children, onSubmit, onClick, className }: ButtonComponent) => {
    return (
      <StyledButton type={type} variant={variant} onSubmit={onSubmit} onClick={onClick} className={className}>
        {children}
      </StyledButton>
    );
  }
)``;

const StyledButton = styled.button<ButtonProps>`
  height: 48px;
  width: 156px;
  border-radius: 12px;
  border: medium none;
  cursor: pointer;
  transition: 0.5s;
  font-style: normal;
  font-weight: 700;
  line-height: 39px;
  font-size: 32px;
  color: ${({ theme, variant }) => theme.colors.control.button[variant].color};
  background: ${({ theme, variant }) => theme.colors.control.button[variant].background};
  font-family: ${({ theme }) => theme.fonts.main};

  &:hover {
    background: ${({ theme, variant }) => theme.colors.control.button[variant].hoverBackground};
  }
`;
