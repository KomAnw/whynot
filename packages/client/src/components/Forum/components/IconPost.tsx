export const IconPost = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M10.67 19.9701C10.43 19.9901 10.19 20.0001 9.95 20.0001C8.78 20.0001 7.66999 19.7201 6.67999
      19.2301L2.78999 19.7901L3.38 15.8801H3.39C2.99 15.0001 2.75 14.0301 2.75 13.0001C2.75 11.1101 3.51999
      9.40014 4.75999 8.14014"
      stroke="#6457B8"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M10.8601 11.7802C11.3019 11.7802 11.6601 11.431 11.6601 11.0002C11.6601 10.5694 11.3019 10.2202
      10.8601 10.2202C10.4182 10.2202 10.0601 10.5694 10.0601 11.0002C10.0601 11.431 10.4182 11.7802 10.8601 11.7802Z"
      fill="#6457B8"
    />
    <path
      d="M14.0603 11.7802C14.5021 11.7802 14.8603 11.431 14.8603 11.0002C14.8603 10.5694 14.5021 10.2202
      14.0603 10.2202C13.6184 10.2202 13.2603 10.5694 13.2603 11.0002C13.2603 11.431 13.6184 11.7802 14.0603
      11.7802Z"
      fill="#6457B8"
    />
    <path
      d="M17.2502 11.7802C17.692 11.7802 18.0502 11.431 18.0502 11.0002C18.0502 10.5694 17.692 10.2202
      17.2502 10.2202C16.8084 10.2202 16.4502 10.5694 16.4502 11.0002C16.4502 11.431 16.8084 11.7802 17.2502 11.7802Z"
      fill="#6457B8"
    />
    <path
      d="M20.6199 13.88L21.2099 17.79L17.3199 17.23C16.6199 17.58 15.8599 17.82 15.0499 17.93C14.7299
      17.98 14.3899 18 14.0499 18C10.0699 18 6.85986 14.87 6.85986 11C6.85986 9.07 7.64986 7.35 8.94986 6.08C10.2399
      4.79 12.0499 4 14.0599 4C18.0399 4 21.2499 7.13 21.2499 11C21.2499 12.03 21.0099 13 20.6099 13.88H20.6199V13.88Z"
      stroke="#6457B8"
      strokeWidth="1.5"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);
