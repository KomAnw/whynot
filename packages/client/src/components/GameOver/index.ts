export { GameOver as default } from './GameOver';
