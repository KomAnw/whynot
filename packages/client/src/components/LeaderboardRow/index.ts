export { LeaderboardRow as default } from './LeaderboardRow';
