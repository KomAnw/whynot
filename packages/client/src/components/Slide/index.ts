export { SlideHistory, SlideTutorial } from './Slide';
