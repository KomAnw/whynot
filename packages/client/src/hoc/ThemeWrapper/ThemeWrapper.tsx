import { ThemeProvider } from 'styled-components';
import { useAppSelector } from 'src/hooks/redux';
import type { ThemeWrapperProps } from './types';

const ThemeWrapper = ({ children }: ThemeWrapperProps) => {
  const theme = useAppSelector(({ theme }) => theme);

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export default ThemeWrapper;
