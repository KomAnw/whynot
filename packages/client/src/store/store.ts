import { configureStore } from '@reduxjs/toolkit';
import { userApi } from 'src/api/user/user';
import { authApi } from 'src/api/auth/auth';
import themeReducer from 'src/hoc/ThemeWrapper/themeSlice';
import modeReducer from 'pages/Game/modeSlice';
import fullscreenReducer from 'pages/Settings/fullscreenSlice';
import gamepadReducer from 'pages/Settings/gamepadSlice';

const store = configureStore({
  reducer: {
    theme: themeReducer,
    mode: modeReducer,
    fullscreen: fullscreenReducer,
    gamepad: gamepadReducer,
    [authApi.reducerPath]: authApi.reducer,
    [userApi.reducerPath]: userApi.reducer,
  },
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(authApi.middleware, userApi.middleware),
});

export default store;

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
