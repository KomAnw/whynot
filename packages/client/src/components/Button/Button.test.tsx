import { Button } from 'components/Button';
import { create } from 'react-test-renderer';
import ThemeWrapper from 'src/hoc/ThemeWrapper';
import { Provider } from 'react-redux';
import store from 'src/store';

describe('Link', () => {
  it('renders <Button /> correctly', () => {
    const mockFn = jest.fn(() => null);
    const tree = create(
      <Provider store={store}>
        <ThemeWrapper>
          <Button variant="primary" type="submit" onClick={mockFn} onSubmit={mockFn}>
            LOGIN
          </Button>
        </ThemeWrapper>
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
