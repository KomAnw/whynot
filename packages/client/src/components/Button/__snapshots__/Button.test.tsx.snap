// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders <Button /> correctly 1`] = `
<button
  className="sc-gswNZR juODTC sc-bcXHqe"
  onClick={[MockFunction]}
  onSubmit={[MockFunction]}
  type="submit"
>
  LOGIN
</button>
`;
